sdk/storage/azure-storage-blob/azure/storage/blob/_shared/policies.py

```diff
@@ -0,0 +1,594 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import base64
+import hashlib
+import re
+import random
+from time import time
+from io import SEEK_SET, UnsupportedOperation
+import logging
+import uuid
+import types
+from wsgiref.handlers import format_date_time
+try:
+    from urllib.parse import (
+        urlparse,
+        parse_qsl,
+        urlunparse,
+        urlencode,
+    )
+except ImportError:
+    from urllib import urlencode
+    from urlparse import (
+        urlparse,
+        parse_qsl,
+        urlunparse,
+    )
+
+from azure.core.pipeline.policies import (
+    HeadersPolicy,
+    SansIOHTTPPolicy,
+    NetworkTraceLoggingPolicy,
+    HTTPPolicy)
+from azure.core.pipeline.policies.base import RequestHistory
+from azure.core.exceptions import AzureError, ServiceRequestError, ServiceResponseError
+
+from .models import LocationMode
+
+try:
+    _unicode_type = unicode
+except NameError:
+    _unicode_type = str
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def encode_base64(data):
+    if isinstance(data, _unicode_type):
+        data = data.encode('utf-8')
+    encoded = base64.b64encode(data)
+    return encoded.decode('utf-8')
+
+
+def is_retry(response, mode):
```

Comment:
Not sure if this Is the duplicate of https://github.com/Azure/azure-sdk-for-python/blob/88eb9bb8d079eb0cf9c7ecc67244841025ffed3f/sdk/core/azure-core/azure/core/pipeline/policies/retry.py#L250 would you like to confirm it ^_^